pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/model/conditions/Failure.groovy

```diff
@@ -46,7 +49,12 @@ class Failure extends BuildCondition {
 
     @Override
     boolean meetsCondition(@Nonnull WorkflowRun r, Object context, Throwable error) {
-        return combineResults(r, error) == Result.FAILURE
+        FlowNode startNode = null
+        FlowNode endNode = null
+        if (context instanceof Stage) {
+            (startNode, endNode) = Utils.findPossiblyUnfinishedEndNodeForCurrentStage(((Stage)context).name, r.getExecution())
+        }
+        return combineResults(r, error, startNode, endNode) == Result.FAILURE
```

Comment:
Refactor to a shared method the returns `boolean`?
```java
        return combinedResultEquals(r, error, context, Result.FAILURE);
```
Or at least: 
```java
        return combineResults(r, error, context) == Result.FAILURE;
```

Reply:
Well, it's gotta be `(r, error, startNode, endNode)` (as explained elsewhere), so I tend towards just leaving it as is.